rust_src/src/alloc.rs

```diff
@@ -0,0 +1,39 @@
+//! Storage allocation and gc
+
+use remacs_macros::lisp_fn;
+use remacs_sys::globals;
+use remacs_sys::{bounded_number, constype, listn};
+
+use lisp::defsubr;
+use lisp::LispObject;
+
+/// Return a list of counters that measure how much consing there has been.
+/// Each of these counters increments for a certain kind of object.
+/// The counters wrap around from the largest positive integer to zero.
+/// Garbage collection does not decrease them.
+/// The elements of the value are as follows:
+///   (CONSES FLOATS VECTOR-CELLS SYMBOLS STRING-CHARS MISCS INTERVALS STRINGS)
+/// All are in units of 1 = one object consed except for VECTOR-CELLS
+/// and STRING-CHARS, which count the total length of objects consed.
+/// MISCS include overlays, markers, and some internal types.
+/// Frames, windows, buffers, and subprocesses count as vectors
+///   (but the contents of a buffer's text do not count here).
+#[lisp_fn]
+pub fn memory_use_counts() -> LispObject {
```

Comment:
Do we not have a way to turn a Vector into the expected list so this is not as hard coded.

Reply:
```rust
pub fn memory_use_counts() -> LispObject {
    unsafe {
        list(&[
            bounded_number(globals.cons_cells_consed),
            bounded_number(globals.floats_consed),
            bounded_number(globals.vector_cells_consed),
            bounded_number(globals.symbols_consed),
            bounded_number(globals.string_chars_consed),
            bounded_number(globals.misc_objects_consed),
            bounded_number(globals.intervals_consed),
            bounded_number(globals.strings_consed),
        ])
    }
}
```

That works, but there is probably a Rustier way.

Reply:
Okay, I redid the commit to do it this way, except that I put the `From` in `lisp.rs`. I tried to do `vec![x,y,z,...].iter().map(bounded_number).collect()`, but there was some kind of error about something not being implemented for `FnMut` something.